### PR TITLE
doc: MCUboot documentation update

### PR DESCRIPTION
**Pull request by FrancescoSer**
Updated:
 readme-ncs.rst
 wrapper.rst
 index-ncs.rst
 unused_files.rst
 doc/mcuboot/conf.py

Added:
 doc/mcuboot/links.txt
 doc/mcuboot/shortcuts.txt

Signed-off-by: Francesco Domenico Servidio <francesco.servidio@nordicsemi.no>

### CODE REVIEW

File: doc/mcuboot/readme-ncs.rst [hunk 1]
```diff
@@ -3,14 +3,30 @@
...
 
-See :doc:`readme-zephyr` for general information on how to integrate MCUboot with Zephyr.
+MCUboot can be used in the |NCS| in one of the following roles:
 
-The nRF Connect SDK provides additional functionality that is available when MCUboot is included.
-This functionality is implemented in the files in the :file:`modules/mcuboot` subfolder in the `sdk-nrf`_ repository.
+* As an :ref:`Immutable first-stage bootloader <immutable_bootloader>` that can perform device firmware updates to the application.
+* As an :ref:`Upgradable second-stage bootloader <upgradable_bootloader>` that can perform device firmware updates to both itself and the application.
+
+The MCUboot files in the |NCS| are located in the :file:`modules/mcuboot` subfolder in the `sdk-nrf`_ repository.
```
Comment (kyberdin) [inline doc/mcuboot/readme-ncs.rst:11]: Should this be clarified a bit to say something like MCUboot-specific integration files for NCS are in this subfolder? The current form sounds a lot like MCUboot itself is located there, but it's a separate module.

Comment (FrancescoSer) [reply doc/mcuboot/readme-ncs.rst]: right, fixed

**Comment by greg-fer** [comment]
@FrancescoSer , is this planned for 2.1.0 doc improvement?

**Comment by FrancescoSer** [comment]
> @FrancescoSer , is this planned for 2.1.0 doc improvement?

No, it is just a general improvement, not planned for any specific release

File: doc/mcuboot/readme-ncs.rst [hunk 2]
```diff
@@ -3,14 +3,26 @@
 Using MCUboot in nRF Connect SDK
 ################################
 
-See :doc:`readme-zephyr` for general information on how to integrate MCUboot with Zephyr.
+MCUboot can be used in the nRF Connect SDK in one of the following roles:
 
-The nRF Connect SDK provides additional functionality that is available when MCUboot is included.
-This functionality is implemented in the files in the :file:`modules/mcuboot` subfolder in the `sdk-nrf`_ repository.
+* As an `Immutable first-stage bootloader <https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_bootloader.html#first-stage-immutable-bootloader>`_ that can perform device firmware updates to the application.
+* As an `Upgradable second-stage bootloader <https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_bootloader.html#second-stage-upgradable-bootloader>`_ that can perform device firmware updates to both itself and the application.
```
Comment (greg-fer) [inline doc/mcuboot/readme-ncs.rst:9]: I think these links can be added to a links.txt in the doc/mcuboot subfolder...

File: doc/mcuboot/readme-ncs.rst [hunk 3]
```diff
@@ -3,14 +3,26 @@
...
+
+The MCUboot-specific integration files for the nRF Connect SDK are located in the :file:`modules/mcuboot` subfolder in the ``sdk-nrf`` repository.
```
Comment (greg-fer) [inline doc/mcuboot/readme-ncs.rst:11]: Link to the repository? Either same way as you linked to developer.nordicsemi.com above or using links.txt for the doc/mcuboot subfolder, if possible.

File: doc/mcuboot/readme-ncs.rst [hunk 4]
```diff
@@ -3,14 +3,26 @@
...
 
 To include MCUboot in your nRF Connect SDK application, enable :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT`.
+MCUboot will be configured as either the first-stage or the second-stage bootloader, when :kconfig:option:`CONFIG_SECURE_BOOT`` is also enabled.
```
Comment (greg-fer) [inline doc/mcuboot/readme-ncs.rst:14]:
```suggestion
When :kconfig:option:`CONFIG_SECURE_BOOT`` is also enabled, MCUboot is configured as either the first-stage or the second-stage bootloader.
```
And if neither is enabled? Line 6 suggests one of the options must be set, this line suggests setting either is optional...

File: doc/mcuboot/readme-ncs.rst [hunk 5]
```diff
@@ -3,14 +3,26 @@
...
+
+See the following userguides for more information on adding, configuring, and testing MCUboot for your application build in the nRF Connect SDK:
```
Comment (greg-fer) [inline doc/mcuboot/readme-ncs.rst:16]:
```suggestion
See the following user guides for more information on adding, configuring, and testing MCUboot for your application build in the nRF Connect SDK:
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
Nice job! I like the edits! Couple of comments for potential improvements.

File: doc/mcuboot/readme-ncs.rst (hunk 2 above)
Comment (gmarull) [inline doc/mcuboot/readme-ncs.rst:9]: hardcoding `latest` links isn't a good idea when not looking at `latest` rendered docs. But I guess there's no other real option unless we want to introduce circular dependencies (BAD) as with nrfxlib/nrf.

File: doc/mcuboot/readme-ncs.rst (hunk 3 above)
Comment (FrancescoSer) [reply doc/mcuboot/readme-ncs.rst]: linked

File: doc/mcuboot/readme-ncs.rst (hunk 4 above)
Comment (FrancescoSer) [reply doc/mcuboot/readme-ncs.rst]: rephrased

File: doc/mcuboot/readme-ncs.rst (hunk 5 above)
Comment (FrancescoSer) [reply doc/mcuboot/readme-ncs.rst]: fixed

File: doc/mcuboot/readme-ncs.rst [hunk 6]
```diff
@@ -3,22 +3,31 @@
...
 
-The nRF Connect SDK provides additional functionality that is available when MCUboot is included.
-This functionality is implemented in the files in the :file:`modules/mcuboot` subfolder in the `sdk-nrf`_ repository.
+You can use MCUboot in the nRF Connect SDK in one of the following roles:
 
-To include MCUboot in your nRF Connect SDK application, enable :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT`.
+* As an `Immutable first-stage bootloader <https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_bootloader.html#first-stage-immutable-bootloader>`_ that can perform device firmware updates to the application, enabling :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT`
+* As an `Upgradable second-stage bootloader <https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_bootloader.html#second-stage-upgradable-bootloader>`_ that can perform device firmware updates to both itself and the application, enabling both :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` and :kconfig:option:`CONFIG_SECURE_BOOT`.
```
Comment (greg-fer) [inline doc/mcuboot/readme-ncs.rst:11]: Maybe, for your consideration:
```suggestion
* As an `Immutable first-stage bootloader <https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_bootloader.html#first-stage-immutable-bootloader>`_ that can perform device firmware updates to the application. 
  Enable :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` to use it in this role.
* As an `Upgradable second-stage bootloader <https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_bootloader.html#second-stage-upgradable-bootloader>`_ that can perform device firmware updates to both itself and the application. 
  Enable :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` and :kconfig:option:`CONFIG_SECURE_BOOT` to use it in this role.
```

**Review by greg-fer — APPROVED** [review]
One nit.

File: doc/mcuboot/readme-ncs.rst (hunk 6 above)
Comment (FrancescoSer) [reply doc/mcuboot/readme-ncs.rst]: fixed

File: doc/mcuboot/readme-ncs.rst (hunk 2 above)
Comment (FrancescoSer) [reply doc/mcuboot/readme-ncs.rst]: implemented links.txt and shortcuts.txt

**Review by gmarull — APPROVED** [review]
lgtm, one nitpick

**Pull request merged**